src/main/java/chess/controller/ChessGame.java

```diff
@@ -1,48 +1,77 @@
 package chess.controller;
 
+import chess.dao.PiecesDao;
+import chess.db.DBConnector;
 import chess.domain.board.ChessBoard;
+import chess.domain.board.ScoreBoard;
+import chess.domain.piece.Team;
 import chess.domain.state.GameState;
+import chess.domain.state.Progress;
 import chess.domain.state.Ready;
 import chess.dto.ChessBoardDto;
+import chess.service.ChessService;
 import chess.view.InputView;
 import chess.view.OutputView;
 
 import java.util.List;
 import java.util.function.Supplier;
 
 public class ChessGame {
+    private static final String STATUS_COMMAND = "status";
+    private static final ScoreBoard scoreBoard = new ScoreBoard();
+
     private final InputView inputView = new InputView();
     private final OutputView outputView = new OutputView();
+    private final ChessService chessService = new ChessService(new PiecesDao(DBConnector.getProductionDB()));
 
     public void run() {
-        ChessBoard chessBoard = generateChessBoard();
-        GameState gameState = new Ready(chessBoard);
+        ChessBoard chessBoard = setChessBoard();
+        GameState gameState = new Ready(chessBoard, chessService);
         outputView.printStartMessage();
 
-        playGame(gameState, chessBoard);
+        Team winner = playGame(gameState, chessBoard);
+        outputView.printResultMessage(winner);
     }
 
-    private ChessBoard generateChessBoard() {
-        ChessBoard chessBoard = new ChessBoard();
-        chessBoard.initialBoard();
-        return chessBoard;
+    private ChessBoard setChessBoard() {
+        if (chessService.hasNoLastGame()) {
+            ChessBoard chessBoard = new ChessBoard();
+            chessBoard.initialBoard();
+            chessService.saveEntireChessBoard(chessBoard);
+            return chessBoard;
+        }
+        return chessService.loadChessBoard();
     }
 
-    private void playGame(GameState gameState, ChessBoard chessBoard) {
+    private Team playGame(GameState gameState, ChessBoard chessBoard) {
         while (!gameState.isEnd()) {
             GameState currentGameState = gameState;
-            gameState = repeatUntilSuccess(() -> playEachTurn(currentGameState));
+            gameState = repeatUntilSuccess(() -> playEachTurn(currentGameState, chessBoard));
 
             printChessBoardInProgress(gameState, chessBoard);
         }
+        return gameState.findWinner();
     }
 
-    private GameState playEachTurn(GameState gameState) {
+    private GameState playEachTurn(GameState gameState, ChessBoard chessBoard) {
         List<String> command = inputView.readCommand();
+
+        if (command.get(0).equals(STATUS_COMMAND) && gameState.getClass().equals(Progress.class)) {
```

Comment:
실제로 진행해보면 한쪽 팀만 계속 진행할 수 있었습니다 ㅎ 
(move b2 b3, move b3 b4, move b4 b5) 
게임도 완성단계여야 하니까 턴을 잘 구현해주세요.
로드 후에 턴이 제대로 기능하는 부분도요

Reply:
그리고 NPE를 방지하기 위해선 항상 null이 아님이 분명한 쪽이 메서드를 사용하도록 구현해주는 게 좋습니다.
```suggestion
        if (STATUS_COMMAND).equals(command.get(0) && Progress.class.equals(gameState.getClass())) {
```

또 class 타입 비교는 언어에서 지원해주는 연산자인 instanceOf 를 사용하는 게 낫고요,

개인적으로는 이렇게 타입 체크를 해주면서 로직을 진행해야 하면 상태패턴을 사용하지 않는 편이 더 낫다고 생각합니다. 😏

Reply:
> 실제로 진행해보면 한쪽 팀만 계속 진행할 수 있었습니다 ㅎ (move b2 b3, move b3 b4, move b4 b5) 게임도 완성단계여야 하니까 턴을 잘 구현해주세요. 로드 후에 턴이 제대로 기능하는 부분도요

turnsDAO 생성해주었고, 상단 PR comment의 init.sql도 함께 수정해두었습니다.

Reply:
> 그리고 NPE를 방지하기 위해선 항상 null이 아님이 분명한 쪽이 메서드를 사용하도록 구현해주는 게 좋습니다.
> 
> 또 class 타입 비교는 언어에서 지원해주는 연산자인 instanceOf 를 사용하는 게 낫고요,
> 
> 개인적으로는 이렇게 타입 체크를 해주면서 로직을 진행해야 하면 상태패턴을 사용하지 않는 편이 더 낫다고 생각합니다. 😏

equals 적용 순서 변경, Instance of 적용 마쳤습니다! (start 입력 전 status 입력 시 예외 처리를 위함)

아래에서 초기화 로직을 Ready State 안으로 옮기면서도 생각한 거지만, 이게 웨지가 말씀하셨던 '이럴 거면 괜히 구현했다...' 싶은 경우 같기도 합니다. 굳이 적용할 필요 없었던 디자인 패턴을 사용하면서 괜히 프로그램이 더 복잡해지고 체크해줘야 하는 분기가 늘어난 기분이어서요. 🧐 아마 컨트롤러에서 해당 작업을 전부 진행해주었다면 클래스 타입 체크를 할 일도 없었겠죠?

제가 밀고 나가기로 결심한 패턴이다 보니 끝까지 책임지고 싶다는 오기 때문에 열심히 리팩토링하고 있긴 하지만(매몰 비용의 위험성 😱) 이럴 때 웨지는 어떤 결정을 내리시는지도 궁금해지네요! 최초 설계가 그리 효율적이지 않았다는 사실을 프로젝트 중후반쯤에 깨달았다면 웨지는 과감하게 지금까지의 구현을 엎고 리팩토링하시는 편인가요? 물론 지금 제 프로젝트는 아주 아담하고 바꾸려고 든다면 얼마 걸리지 않기야 하겠지만요... ㅎㅎ 🫣

Reply:
만약에 최초 구현이면 다시 엎을수도 있는데요, 
이미 출시된 기능이면 고려해야할게 많아집니다 (변경은 곧 장애 가능성 이거든요 ㅠ)
보통 안 고치고 마음속에 부채1로 남게 되죠..
그렇게 살아가는 거 같습니다